bundles/org.eclipse.swt/Eclipse SWT PI/gtk/org/eclipse/swt/internal/gtk/OS.java

```diff
@@ -86,13 +86,15 @@ public class OS extends C {
 			try {
 				Library.loadLibrary("swt-pi4");
 			} catch (Throwable e) {
+				System.err.println(e.getMessage());
 				System.err.println("SWT OS.java Error: Failed to load swt-pi4, loading swt-pi3 as fallback.");
 				Library.loadLibrary("swt-pi3");
 			}
 		} else {
 			try {
 				Library.loadLibrary("swt-pi3");
 			} catch (Throwable e) {
+				System.err.println(e.getMessage());
```

Comment:
As message could be null, should we check for that case or simply use
```suggestion
				System.err.println(e);
```

Reply:
My preference would be to actually capture the original exception, try the fallback;  If that fails do the `println(e)` of the fallback exception and then throw the original exception to the caller.

Reply:
Please amend this PR accordingly as we should merge the other one ASAP.

Reply:
Done, ready to merge assuming it builds successfully.

Reply:
Great, can you post a how such stacktrace would look like?